type_lens/type_view.py

```diff
@@ -219,14 +219,14 @@ def is_subtype_of(self, typ: Any | tuple[Any, ...], /) -> bool:
             Whether the annotation is a subtype of the given type(s).
         """
         if self.origin:
-            if self.origin in UNION_TYPES:
+            if self.is_union:
                 return all(t.is_subtype_of(typ) for t in self.inner_types)
 
-            return self.origin not in UNION_TYPES and issubclass(self.origin, typ)
+            return issubclass(self.origin, typ)
 
         if self.annotation is AnyStr:
-            return issubclass(str, typ) or issubclass(bytes, typ)
-        return self.annotation is not Any and not self.is_type_var and issubclass(self.annotation, typ)
+            return TypeView(Union[str, bytes]).is_subtype_of(typ)
```

Comment:
this just seems easier to grok to me, and utilizes the API more recursively (which i think is a good thing)